Inc migration from Framework to ASP.NET Core /1 (#27566)
dotnet/AspNetCore.Docs: aspnetcore/migration/inc/usage_guidance.md

**Recommendation**: To enable this on all MVC endpoints, there is an extension method that can be used as follows:

```cs
app.MapDefaultControllerRoute()
    .RequireSystemWebAdapterSession();
```

This also requires some implementation of a session store. For details of options here, see [here](xref:migration/inc/session).

## Remote session exposes additional endpoint for application

The [remote session support](xref:migration/inc/remote-session) exposes an endpoint that allows the core app to retrieve session information. This may cause a potentially long-lived request to exist between the core app and the framework app, but will time out with the current request or the session timeout (by default is 20 minutes).

**Recommendation**: Ensure the API key used is a strong one and that the connection with the framework app is done over SSL.

## Virtual directories must be identical for framework and core applications

The virtual directory setup is used for route generation, authorization, and other services within the system. At this point, no reliable method has been found to enable different virtual directories due to how ASP.NET Framework works.

> **Recomendation**: If you are using virtual directories, ensure your two applications are on different sites with the same application/virtual directory layout.

Reviewer: ```suggestion
**Recomendation**: Ensure your two applications are on different sites (hosts and/or ports) with the same application/virtual directory layout.
```
